backend/Data/Entities/CodeEntity.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+
+namespace Pims.Api.Data.Entities
+{
+    /// <summary>
+    /// CodeEntity class, provides an entity for the datamodel to manage entities that represent codified values.
+    /// </summary>
+    public abstract class CodeEntity : BaseEntity
+    {
+        #region Properties
+        /// <summary>
+        /// get/set - A unique code for the code.
+        /// </summary>
+        /// <value></value>
+        public int Id { get; set; }
+
+        /// <summary>
+        /// get/set - The name of the code.
+        /// </summary>
+        /// <value></value>
+        public string Name { get; set; }
+
+        /// <summary>
+        /// get/set - Whether this code is disabled.
+        /// </summary>
+        /// <value></value>
+        public bool IsDisabled { get; set; }
+
+        public abstract string Type { get;  }
```

Comment:
I think I went a different direction then what you are thinking. This isn't a table, just a different base class that exposes some common properties useful for getting a list of codes. When this base class is applied to an existing entity, you can now use the automapper to get a list of just the Code Entity fields.

For example, in the case of Agency. Instead of getting a list of all the agency specific fields, the CodeEntity Automapper just returns the name, id, isDisabled, and type fields. There is no CodeEntity table. CodeEntity is just a wrapper on top of BaseEntity.

The type field may not be necessary, but is currently used to determine what code set a code belongs to. This is required for the mixed list of code sets returned by the /all LookupCodeController endpoint. from that endpoint you get something like this:

[{
name: MOH
id:1
isDisabled: false
type: agency
},
{
name: Surplus Active
id: 1
isDisabled: false,
type: propertyClassification
}]